R/plot_plate.R

```diff
@@ -0,0 +1,273 @@
+  title <- paste("Counts for", analyte_name)
+
+  plot_plate(colors, plot_title = title, plot_numbers = plot_counts, numbers = counts, plot_legend = plot_legend, legend_mapping = color_map)
+}
+
+
+#' Plot layout of a 96-well plate
+#'
+#' It is a function to plot the layout of a 96-well plate using a color scale.
+#' The function uses the plot_plate function to plot the layout.
+#'
+#' @param plate The plate object with the layout information
+#' @param plot_legend Logical indicating if the legend should be plotted
+#'
+#' @return A ggplot object
+#'
+#' @examples
+#' plate_filepath <- system.file("extdata", "CovidOISExPONTENT_CO.csv", package = "PvSTATEM", mustWork = TRUE)
+#' layout_filepath <- system.file("extdata", "CovidOISExPONTENT_CO_layout.xlsx", package = "PvSTATEM", mustWork = TRUE)
+#' plate <- read_luminex_data(plate_filepath, layout_filepath)
+#' plot_layout(plate = plate, plot_legend = T)
+#'
+#' @export
+plot_layout <- function(plate, plot_legend = TRUE) {
+
+  if (is.null(plate)) {
+    stop("The plate object must be provided")
+  }
+
+  plate_name <- plate$plate_name
+  sample_types <- plate$sample_types
+  location <- plate$sample_locations
+  sample_types <- create_vector_without_holes(sample_types, location)
+
+  if (length(sample_types) != 96) {
+    stop("The sample_types vector must have 96 elements")
+  }
+
+  # Define the mapping using a named vector
+  color_map <- c(
+    "BLANK" = "#B5CFB7",
+    "POSITIVE CONTROL" = "#B1AFFF",
+    "NEGATIVE CONTROL" = "#FFE9D0",
+    "TEST" = "#BBE9FF",
+    "STANDARD CURVE" = "#F7B5CA",
+    " " = "white"
+  )
+
+  # Mapping function using the named vector
+  map_to_color <- function(sample_type) {
+    if (!is.null(color_map[sample_type])) {
+      return(color_map[sample_type])
+    } else {
+      return(color_map[" "])
+    }
+  }
+
+  # Apply the mapping function to the sample_types vector
+  colors <- sapply(sample_types, map_to_color)
+
+  title <- paste("Layout of", plate_name)
+
+  plot_plate(colors, plot_title = title, plot_numbers = FALSE, plot_legend = plot_legend, legend_mapping = color_map)
+}
+
+create_vector_without_holes <- function(vector, locations) {
+  # Create a vector with all the locations
+  all_locations <- c("A1", "A2", "A3", "A4", "A5", "A6", "A7", "A8", "A9", "A10", "A11", "A12",
+                     "B1", "B2", "B3", "B4", "B5", "B6", "B7", "B8", "B9", "B10", "B11", "B12",
+                     "C1", "C2", "C3", "C4", "C5", "C6", "C7", "C8", "C9", "C10", "C11", "C12",
+                     "D1", "D2", "D3", "D4", "D5", "D6", "D7", "D8", "D9", "D10", "D11", "D12",
+                     "E1", "E2", "E3", "E4", "E5", "E6", "E7", "E8", "E9", "E10", "E11", "E12",
+                     "F1", "F2", "F3", "F4", "F5", "F6", "F7", "F8", "F9", "F10", "F11", "F12",
+                     "G1", "G2", "G3", "G4", "G5", "G6", "G7", "G8", "G9", "G10", "G11", "G12",
+                     "H1", "H2", "H3", "H4", "H5", "H6", "H7", "H8", "H9", "H10", "H11", "H12")
```

Comment:
There are most surly more reasonable ways to do that. That actually is more visual, but also will be a struggle to change when we will add more plate types.

Dropping some R code here
```r
a <- rep(LETTERS[1:8], each = 12)
b <- as.character(1:12)
paste0(a, b)
```